app/models/spree/order_decorator.rb

```diff
@@ -0,0 +1,10 @@
+module Spree
+  module OrderDecorator
+    def self.prepended(base)
+      base.insert_checkout_step(:payment_confirm, after: :confirm, if: ->(order) { order.payments.map { |p| p.payment_method&.has_preference?(:intents) && p.payment_method&.get_preference(:intents) }.any? }) unless base.checkout_steps.has_key?(:payment_confirm)
```

Comment:
```suggestion
      base.insert_checkout_step(:payment_confirm, before: :complete, if: ->(order) { order.payments.valid.map { |p| p.payment_method&.has_preference?(:intents) && p.payment_method&.get_preference(:intents) }.any? }) unless base.checkout_steps.has_key?(:payment_confirm)
```

`confirm` is optional so it's safer to put this before `complete`